snuba/datasets/errors.py

```diff
@@ -0,0 +1,302 @@
+from datetime import timedelta
+from typing import Mapping, Sequence, Union
+
+from snuba.clickhouse.columns import (
+    Array,
+    ColumnSet,
+    DateTime,
+    FixedString,
+    IPv4,
+    IPv6,
+    LowCardinality,
+    Materialized,
+    Nested,
+    Nullable,
+    String,
+    UInt,
+    UUID,
+    WithCodecs,
+    WithDefault,
+)
+from snuba.datasets.dataset import ColumnSplitSpec, TimeSeriesDataset
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.table_storage import TableWriter, KafkaStreamLoader
+from snuba.datasets.errors_processor import ErrorsProcessor
+from snuba.datasets.schemas.tables import ReplacingMergeTreeSchema
+from snuba.datasets.tags_column_processor import TagColumnProcessor
+from snuba.query.processors.basic_functions import BasicFunctionsProcessor
+from snuba.query.processors.prewhere import PrewhereProcessor
+from snuba.query.query import Query
+from snuba.query.extensions import QueryExtension
+from snuba.query.parsing import ParsingContext
+from snuba.query.query_processor import QueryProcessor
+from snuba.query.timeseries import TimeSeriesExtension
+from snuba.query.project_extension import ProjectExtension, ProjectWithGroupsProcessor
+
+
+class ErrorsDataset(TimeSeriesDataset):
+    """
+    Represents the collections of all event types that are not transactions.
```

Comment:
Is it worthwhile splitting out storage for CSP events or are they similar enough to errors that and low volume enough that it doesn't matter?

Reply:
Not at this time at least. They are a tiny amount compared to errors.

Reply:
Sounds good.